### PR TITLE
Reporting Enso test failures as annotations on the PR

### PR DESCRIPTION
**Pull request by radeusgd**
### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [ ] Unit tests have been written where possible.

### CODE REVIEW

**Comment by radeusgd** [comment]
image

This is what the annotations look like on a PR. I think this could be a helpful improvement when looking for test failures - instead of having to dig into the workflows, we can display them right there.

They also appear in the workflow summary:
image

File: build/build/src/enso.rs [hunk 1]
```diff
@@ -223,7 +223,9 @@ impl BuiltEnso {
         if errors.is_empty() {
             Ok(())
         } else {
-            error!("{} test suit(s) failed.", errors.len());
+            let summary = errors.as_slice().iter().map(|e| e.to_string()).collect::<Vec<_>>().join(", ");
+            println!("::error title=Failed Standard Library Tests::{} test suite(s) failed: {}", errors.len(), summary);
```
Comment (JaroslavTulach) [inline build/build/src/enso.rs]: I see a line starting with `::` - that must be a _GitHub Actions command line_! Where is the syntax documented?

Comment (JaroslavTulach) [reply build/build/src/enso.rs]: I am glad we ended up doing the same - e.g. using `println!` to print the command line without a prefix. Just like at [my PR](https://github.com/enso-org/enso/pull/10914/files#r1738221228).

**Comment by radeusgd** [comment]
I did not manage to cleanup and finish this yet because I was focusing on more pressing tasks. I will re-visit this once I'm back from vacation.

**Review by GregoryTravis — APPROVED** [review]

**Comment by radeusgd** [comment]
After the changes now the folding of benchmarks is working again and now we get a nice list of test failures in the checks and files changed panels, next to relevant test code.

**Review by marthasharkey — APPROVED** [review]

**Review by Frizi — APPROVED** [review]

**Pull request merged**